istioctl/pkg/writer/envoy/configdump/configdump.go

```diff
@@ -115,22 +115,31 @@ func (c *ConfigWriter) PrintSecretSummary() error {
 	return secretWriter.PrintSecretItems(secretItems)
 }
 
-func (c *ConfigWriter) PrintFullSummary(cf ClusterFilter, lf ListenerFilter, rf RouteFilter) error {
+func (c *ConfigWriter) PrintFullSummary(cf ClusterFilter, lf ListenerFilter, rf RouteFilter, epf EndpointFilter) error {
+	w := c.Stdout
+	writeSeperator := func(writer io.Writer, configName string) {
+		fmt.Fprintf(writer, "\n--------------------\n%s CONFIG:\n\n", configName)
```

Comment:
Its kind of weird the first line of the ouutput is:

```

--------------------
CLUSTER CONFIG:

```

including newlines, etc.

This isn't how k8s works btw:
```
$ k get svc,pod
NAME                    TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                     AGE
service/details         ClusterIP   10.96.67.208    <none>        9080/TCP                                    6d22h
service/echo-v1         ClusterIP   10.96.9.154     <none>        80/TCP,443/TCP,9090/TCP,9091/TCP,7070/TCP   15h

NAME                                  READY   STATUS    RESTARTS   AGE
pod/details-v1-6997d94bb9-lvhw4       2/2     Running   0          15h
pod/echo-v1-7f68b8c7c4-ppdt9          2/2     Running   0          15h
```

Reply:
I see, the k8s format looks better

Reply:
Revised to format like below:
```
 istioctl pc all bookinfo-reviews-istio-waypoint-8584698898-qmzcm.default    
NAME                                                                   SERVICE FQDN                             PORT     SUBSET      DIRECTION       TYPE             DESTINATION RULE
cluster/agent                                                          cluster/agent                            -        -           -               STATIC           
cluster/connect_originate                                              cluster/connect_originate                -        -           -               ORIGINAL_DST     
cluster/encap                                                          cluster/encap                            -        -           -               STATIC           
cluster/inbound-vip|443|tcp|kubernetes.default.svc.cluster.local       kubernetes.default.svc.cluster.local     443      tcp         inbound-vip     EDS              
cluster/main_internal                                                  cluster/main_internal                    -        -           -               STATIC           
cluster/prometheus_stats                                               cluster/prometheus_stats                 -        -           -               STATIC           
cluster/inbound-vip|9080|http|reviews.default.svc.cluster.local        reviews.default.svc.cluster.local        9080     http        inbound-vip     EDS              
cluster/inbound-vip|9080|http/v1|reviews.default.svc.cluster.local     reviews.default.svc.cluster.local        9080     http/v1     inbound-vip     EDS              
cluster/inbound-vip|9080|http/v2|reviews.default.svc.cluster.local     reviews.default.svc.cluster.local        9080     http/v2     inbound-vip     EDS              
cluster/inbound-vip|9080|http/v3|reviews.default.svc.cluster.local     reviews.default.svc.cluster.local        9080     http/v3     inbound-vip     EDS              
cluster/sds-grpc                                                       cluster/sds-grpc                         -        -           -               STATIC           
cluster/xds-grpc                                                       cluster/xds-grpc                         -        -           -               STATIC           
cluster/zipkin                                                         cluster/zipkin                           -        -           -               STRICT_DNS       

NAME                                              ADDRESS PORT  MATCH DESTINATION
listener/main_internal                                    0     ALL   Inline Route: /*
listener/listener/main_internal                           0     ALL   Cluster: inbound-vip|443|tcp|kubernetes.default.svc.cluster.local
listener/listener/listener/main_internal                  0     ALL   Cluster: encap
listener/listener/listener/listener/main_internal         0     ALL   Cluster: encap
listener/connect_originate                                0     ALL   Cluster: connect_originate
listener/connect_terminate                        0.0.0.0 15008 ALL   Inline Route: 
listener/                                         0.0.0.0 15021 ALL   Inline Route: /healthz/ready*
listener/                                         0.0.0.0 15090 ALL   Inline Route: /stats/prometheus*

NAME                                                              DOMAINS     MATCH                  VIRTUAL SERVICE
route/                                                            *           /healthz/ready*        
route/default                                                     *                                  
route/inbound-vip|9080|http|reviews.default.svc.cluster.local     *           /*                     reviews.default
route/                                                            *           /stats/prometheus*     
route/encap                                                       *           /*                     

RESOURCE NAME      TYPE           STATUS     VALID CERT     SERIAL NUMBER                               NOT AFTER                NOT BEFORE
secret/default     Cert Chain     ACTIVE     true           144021931742737826617984529057324126332     2023-03-10T06:44:41Z     2023-03-09T06:42:41Z
secret/ROOTCA      CA             ACTIVE     true           283261770163279550791033412368149688088     2033-02-25T05:34:38Z     2023-02-28T05:34:38Z

NAME                                                      STATUS      LOCALITY     CLUSTER
endpoint/127.0.0.1:15020                                  HEALTHY                  agent
endpoint/envoy://connect_originate/                       HEALTHY                  encap
endpoint/envoy://main_internal/                           HEALTHY                  main_internal
endpoint/127.0.0.1:15000                                  HEALTHY                  prometheus_stats
endpoint/envoy://connect_originate/10.244.2.69:9080       HEALTHY                  inbound-vip|9080|http|reviews.default.svc.cluster.local
endpoint/envoy://connect_originate/10.244.2.68:9080       HEALTHY                  inbound-vip|9080|http|reviews.default.svc.cluster.local
endpoint/envoy://connect_originate/10.244.2.70:9080       HEALTHY                  inbound-vip|9080|http|reviews.default.svc.cluster.local
endpoint/envoy://connect_originate/10.244.2.69:9080       HEALTHY                  inbound-vip|9080|http/v1|reviews.default.svc.cluster.local
endpoint/envoy://connect_originate/10.244.2.68:9080       HEALTHY                  inbound-vip|9080|http/v2|reviews.default.svc.cluster.local
endpoint/envoy://connect_originate/10.244.2.70:9080       HEALTHY                  inbound-vip|9080|http/v3|reviews.default.svc.cluster.local
endpoint/./var/run/secrets/workload-spiffe-uds/socket     HEALTHY                  sds-grpc
endpoint/./etc/istio/proxy/XDS                            HEALTHY                  xds-grpc
```

Single config command like `pc cluster`/`pc listener` won't be affected. For example:
```
istioctl pc cluster bookinfo-reviews-istio-waypoint-8584698898-qmzcm.default
SERVICE FQDN                             PORT     SUBSET      DIRECTION       TYPE             DESTINATION RULE
agent                                    -        -           -               STATIC           
connect_originate                        -        -           -               ORIGINAL_DST     
encap                                    -        -           -               STATIC           
kubernetes.default.svc.cluster.local     443      tcp         inbound-vip     EDS              
main_internal                            -        -           -               STATIC           
prometheus_stats                         -        -           -               STATIC           
reviews.default.svc.cluster.local        9080     http        inbound-vip     EDS              
reviews.default.svc.cluster.local        9080     http/v1     inbound-vip     EDS              
reviews.default.svc.cluster.local        9080     http/v2     inbound-vip     EDS              
reviews.default.svc.cluster.local        9080     http/v3     inbound-vip     EDS              
sds-grpc                                 -        -           -               STATIC           
xds-grpc                                 -        -           -               STATIC           
zipkin                                   -        -           -               STRICT_DNS 
```

WDYT?

Reply:
I like it!  Thanks @hanxiaop for working on this!!